File: 3024_Type_of_Triangle/solution.c
int comp(const void* a, const void* b) { return (*(int*)a - *(int*)b); }
char* triangleType(int* nums, int numsSize) {
    qsort(nums, numsSize, sizeof(nums[0]), comp);
    if (nums[0] + nums[1] <= nums[2]) {
        return "none";
    } else if (nums[0] == nums[1] && nums[1] == nums[2]) {
        return "equilateral";
    } else if (nums[0] != nums[1] && nums[0] != nums[2] && nums[1] != nums[2]) {
        return "scalene";
    } else {
        return "isosceles";
    }
}

File: 3024_Type_of_Triangle/solution.py
class Solution:
    def triangleType(self, nums: List[int]) -> str:
        nums.sort()
        if nums[0] + nums[1] <= nums[2]:
            return "none"
        elif all(num == nums[0] for num in nums):
            return "equilateral"
        elif nums[0] != nums[1] and nums[0] != nums[2] and nums[1] != nums[2]:
            return "scalene"
        else:
            return "isosceles"
